Model/Builders/RecurringHelper.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace GingerPay\Payment\Model\Builders;
+
+use GingerPay\Payment\Model\OrderCollection\Orders;
+use GingerPay\Payment\Model\Api\UrlProvider;
+use GingerPay\Payment\Model\Builders\MailTransportBuilder;
+use Magento\Sales\Api\Data\OrderInterface;
+
+class RecurringHelper
+{
+    /**
+     * @var Orders
+     */
+    public $orders;
+    /**
+     * @var UrlProvider
+     */
+    public $urlProvider;
+    /**
+     * @var MailTransportBuilder
+     */
+    protected $mailTransport;
+
+    /**
+     * RecurringHelper constructor.
+     *
+     * @param Orders                $orders
+     * @param UrlProvider           $urlProvider
+     * @param MailTransportBuilder  $mailTransport
+     */
+    public function __construct(
+        Orders                  $orders,
+        UrlProvider             $urlProvider,
+        MailTransportBuilder    $mailTransport
+    ) {
+        $this->orders = $orders;
+        $this->urlProvider = $urlProvider;
+        $this->mailTransport = $mailTransport;
+    }
+
+    public function initializeRecurringOrder($order, $isRecurringEnable)
+    {
+        $additionalData = $order->getPayment()->getAdditionalInformation();
+
+        if (!empty($additionalData['periodicity']) && $additionalData['periodicity'] != 'once' && $isRecurringEnable)
+        {
+            $recurringType = 'first';
+            $time = strtotime(date('Y-m-d H:i'));
+            $recurringPeriodicity = $additionalData['periodicity'];
+            $this->orders->saveInitializeData(
+                $order,
+                [
+                    'next_payment_date' => $this->getNextPaymentDate($time, $recurringPeriodicity),
+                    'recurring_periodicity' => $recurringPeriodicity
+                ]
+            );
+        }
+    }
+
+
+    public function getNextPaymentDate($currentDate, $recurringPeriodicity)
+    {
+        return strtotime($recurringPeriodicity, $currentDate);
+    }
+
+    public function isItFirstRecurringTransaction($transaction) : bool
+    {
+        if (empty(current($transaction['transactions'])["payment_method_details"]["recurring_type"]))
+        {
+            return false;
+        }
+        else
```

Comment:
Body of the if statement contains return, so code after it will be not processing if statement is true and will be processing if statement is false, **`else` there do not require**.